torchprime/torch_xla_models/optimization/checkpointing.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from functools import partial
+
+import torch_xla.distributed.spmd as xs
+from torch_xla.distributed.fsdp import checkpoint_module
+from transformers.trainer_pt_utils import get_module_class_from_name
+
+from torchprime.layers.sequential import HomogeneousSequential
+from torchprime.sharding.shard_model import wrap_module
+from torchprime.torch_xla_models import offloading, remat_all, scan_layers
+
+logger = logging.getLogger(__name__)
+
+
+def add_activation_checkpointing_and_scan(model, config):
```

Comment:
add docstring and type annotations for public methods, that's the general Python guideline we should follow